src/http/common.cr

```diff
@@ -40,6 +40,10 @@ module HTTP
           body = UnknownLengthContent.new(io)
         end
 
+        if body.is_a?(Content) && expect_continue?(headers)
```

Comment:
`body` should be a `Content` here in any case. IMO restricting the type with `body : Content? = nil` above is cleaner.

Reply:
Unfortunately this doesn't work. The assignment `body = ChunkedContent.new(io)` then fails with `type must be (HTTP::Content | Nil), not (IO | Nil)` :(

Reply:
Oh, that looks like an error in the type inference. `ChunkedContent.new(io)` is actually a `HTTP::Content`. And it can't be `Nil`.

Reply:
> And it can't be `Nil`.

@straight-shoota there's `body = nil` on line 32, that's the reason I guess.

Reply:
Ah, yes. It means the resulting type of `body`. I interpreted as trying to assign `IO | Nil` to the variable which is restricted to `Content | Nil`. But yeah, that makes sense.